{
  MultiException me=new MultiException();
  IOException io=new IOException("one");
  me.add(io);
  assertEquals(1,me.size());
  try {
    me.ifExceptionThrow();
    assertTrue(false);
  }
 catch (  IOException e) {
    assertTrue(e == io);
  }
  try {
    me.ifExceptionThrowMulti();
    assertTrue(false);
  }
 catch (  MultiException e) {
    assertTrue(e == me);
  }
  try {
    me.ifExceptionThrowRuntime();
    assertTrue(false);
  }
 catch (  RuntimeException e) {
    assertTrue(e.getCause() == io);
  }
  me=new MultiException();
  RuntimeException run=new RuntimeException("one");
  me.add(run);
  try {
    me.ifExceptionThrowRuntime();
    assertTrue(false);
  }
 catch (  RuntimeException e) {
    assertTrue(run == e);
  }
}
